### ISSUE TITLE
Consider using cage for gnome/firefox less execution

### ISSUE THREAD

**Issue by lkocman** [yast/d-installer#341]
using something like https://github.com/Hjdskes/cage for d-installer would eliminate the requirement to load gnome and firefox 

Originally raised by Patrick Fitzgerald on the community call.

**Comment by ancorgs** [comment]
We welcome any idea in these fields:

- An improved and more light-weight Live ISO.
- A container that can run Wayland + a browser so we don't even need an ISO (see some experiments [here](https://build.opensuse.org/package/show/YaST:Head:Containers/kiosk-container)

Those are areas in which experts in operating systems, graphical environments, browsers and/or containers could help without needing to code D-Installer or YaST.

If you are reading this, please consider jumping in. :wink:

**Comment by lkocman** [comment]
Communicated back to Patrick, just for tracking you could start with forking this project
https://build.opensuse.org/package/show/YaST:Head:D-Installer/d-installer-live

**Comment by jreidinger** [comment]
Yeah, I play with cage and also sway ( which is in TW as opposed to cage ) and my main issue was that it failing with Leap15.X qemu. So can be tested only on TW host which I sadly do not have at hand. And question is how looks real hardware support ( especially on other archs like aarch64 ).

**Comment by javierllorente** [comment]
Perhaps we could get rid of Firefox completely by using an embedded Web browser (ie: Qt WebEngine)

**Comment by jotolo2** [comment]
Hi guys!
I've been able to configure a live `d-installer` iso that uses `cage` with a `qt-kiosk-browser`:
https://build.opensuse.org/package/show/home:jtorres:branches:YaST:Head:D-Installer/d-installer-live-test

This is a POC and many things should be improved, more specifically on `cage` side.

`Cage` is started as a systemd unit service. At this moment it can display the browser with `cockpit-desktop`, however, it has some issues with the cursor and a lot of lagging, it's not usable. I should research what is making this issue. 

It can be attached to that `cage` instance a `wayvnc` daemon to show its display through VNC protocol. I've set up a systemd unit service for this too.

The `qt-kiosk-browser` is a minimal browser I've set up based on the `Qt` examples. It would make sense to see how to make it more minimal and avoid some `Qt` dependencies it has at the moment.

The next step I will do is drop all Xorg dependencies and see if the iso improves on light-weight.

**Comment by Lunarequest** [comment]
Hi I just had a conversation with @lkocman about cockpit and this was brought up there. I was wondering has anyone looked into webkit2gtk(3/4) as a replacement for qt-kiosk-browser.

There are some caveats such as hardware acceleration on NVIDIA cards being broken and hence requiring `WEBKIT_DISABLE_DMABUF_RENDERER` to be set to 1, this causes some artifacts in rendering.

**Comment by ancorgs** [comment]
> I was wondering has anyone looked into webkit2gtk(3/4)

Yes, I have briefly looked into it. I really like its potential as a kiosk browser and the source code is small and understandable.

But I'm pretty clueless regarding the rendering artifacts. @joantolo may know more details.

**Comment by Lunarequest** [comment]
in my experience the artifacts are effectively elements that were removed/updated ghosting and staying behind. Until the browser window was forcefully resized

**Comment by imobachgs** [comment]
About using Firefox instead of Electron or Tauri, I do not think it is "the wrong choice". It is just another choice :-) Actually, the plan is to use Firefox Kiosk (but it has not happened yet due to some problems with Firefox and Wayland, although I think they are fixed already).

We considered both of them, but we could not find the time to do an in-depth research. However, we need to have a few things into account:

* Both of them will rely on a web engine with JavaScript support. I do not expect a big impact on performance compared to Firefox. But we do not have numbers.
* We need Electron/Tauri to be fully supported in SUSE Linux Enteprise Server for all the architectures.

We might consider something like that in the future, but as mentioned before, we are focused in other areas and we do not have the time to work on this.
